### ISSUE TITLE
[Informix] Query to long when execute  selectRoutineParametersSql

### ISSUE THREAD

**Issue by jonnyforestGIS** [schemaspy/schemaspy#1472]
Hi hope this is useful to update Informix provider. @npetzall 

* Full command including arguments how you run SchemaSpy:

```cmd

java -jar schemaspy.jar -configFile informix.properties  -debug 

```
I noticed that takes too long to get the query related to selectRoutineParameterSQL. I'm using a QA Informix 14.10 db with a considerable number of routines (more than 1200).

## Expected Behavior

Fast query to selectRoutineParameterSQL to retrieve data.

## Current Behavior

I reproduce the query from informix.properties [L32C1-L32C2] (https://github.com/schemaspy/schemaspy/blob/010da21ce8b0ecc00d75babe933f4a57f22df36f/src/main/resources/org/schemaspy/types/informix.properties#L32C1-L32C2 )

```sql

select
	p.procname || '(' || ifx_param_types(p.procid) || ')' as routine_name,
	case
		p.isproc when "f" then "function"
		when "t" then "procedure"
	end as routine_type,
	ifx_ret_types(p.procid) as dtd_identifier,
	l.langname as routine_body,
	b.data as routine_definition,
	b.seqno,
	'N/A' as sql_data_access,
	case
		LOWER (p.mode) when "d" then "dba"
		when "o" then "Owner"
		when "p" then "Protected"
		when "r" then "Restricted"
		when "t" then "trigger"
	end as security_type,
	p.variant as is_deterministic,
	"N / A" as routine_comment
from
	sysprocedures p
join sysroutinelangs l on
	p.langid = l.langid
join sysprocbody b on
	p.procid = b.procid
	and b.datakey = 'T'
order by
	p.procid,
	b.seqno

```

The query makes schemaspy fail.

## Possible Solution

I optimize the query using a sub-query and I was able to reduce endless minutes without retrieving data from the original query to milliseconds. The problem was in the two functions ifx_param_types() and ifx_ret_types().

So the optimised query is:

```sql

select
	procname || '(' || ifx_param_types(procid) || ')' as routine_name,
	routine_type,
	ifx_ret_types(procid) as dtd_identifier,
	routine_body,
	routine_definition,
	seqno,
	sql_data_access,
	security_type,
	is_deterministic,
	routine_comment
	
from
	(
	select
		p.procid,
	    p.procname,
		p.langid,
		case
			p.isproc when "f" then "function"
			when "t" then "procedure"
		end as routine_type,
		l.langname as routine_body,
		b.data as routine_definition,
		b.seqno,
		'N/A' as sql_data_access,
		case
			LOWER (p.mode) when "d" then "dba"
			when "o" then "Owner"
			when "p" then "Protected"
			when "r" then "Restricted"
			when "t" then "trigger"
		end as security_type,
		p.variant as is_deterministic,
		"N / A" as routine_comment
	from
		sysprocedures p
	join sysroutinelangs l on
		p.langid = l.langid
	join sysprocbody b on
		p.procid = b.procid
		and b.datakey = 'T'
	order by
		p.procid,
		b.seqno
	) t

```

## Steps to Reproduce (for bugs)

NA

## Context

* Database type (e.g. MySQL): IBM Informix 14.10
* Database driver version (is you use a special one): https://mvnrepository.com/artifact/com.ibm.informix/jdbc/4.50.10
* Your DDL (sample how to reproduce the issue): NA

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39): NA
* Operating System and version (e.g. Windows 10): Windows 10
* Java version: NA

**Comment by jonnyforestGIS** [comment]
@npetzall thank you for your comment. I don't mean to have creds or fame. I always help and did some PR in the past in others open source projects. But I always open to discussing and learn. In this case, probably the right place to do this is in the discussions. I'm sorry for that. 

I tagged you because of the commits that you have related to the driver. Can you please help me or can you point some other person in the project to guide me to change the necessary files to add this to a PR or is only necessary change this file?

Thank you for the work in the Informix provider.

Best regards

**Comment by npetzall** [comment]
Absolutely, things are better done in issue first. Avoids discussions in PR.

Nothing to be sorry about. I just don't want you to miss out on recognition. So make it a PR was my approval and nothing else.

You only need to modify that single file.

If you want to test it out first.

You can create a new file let's call it `my_informix.properties` and place it next to the SchemaSpy jar-file. Now you can either extend the existing Informix och copy all of the content from informix.propertied to `my_informix.properties` and run SchemaSpy with `-t my_informix` then you can make modifications to `my_informix.properties` and they will be picked up without needing to rebuild och change SchemaSpy. When everything works as expected. You can just do an online edit in the file and create a PR.
